samtranslator/schema/schema.py

```diff
@@ -45,42 +45,84 @@ class AwsServerlessConnector(BaseModel):
     Properties: ConnectorProperties
 
 
+class FunctionCodeUri(BaseModel):
+    Bucket: Union[str, SamIntrinsic]
+    Key: Union[str, SamIntrinsic]
+    Version: Optional[Union[str, SamIntrinsic]]
+
+
+class FunctionDeploymentPreferenceHooks(BaseModel):
+    PostTraffic: Optional[Union[str, SamIntrinsic]]
+    PreTraffic: Optional[Union[str, SamIntrinsic]]
+
+
+class FunctionDeploymentPreference(BaseModel):
+    Alarms: Optional[Union[List[SamIntrinsic], SamIntrinsic]]
+    Enabled: Optional[Union[bool, SamIntrinsic]]
+    Hooks: Optional[FunctionDeploymentPreferenceHooks]
+    PassthroughCondition: Optional[Union[bool, SamIntrinsic]]
+    Role: Optional[Union[str, SamIntrinsic]]
+    TriggerConfigurations: Optional[PassThrough]
+    Type: Union[str, SamIntrinsic]
+
+
+class FunctionDeadLetterQueue(BaseModel):
+    TargetArn: str
+    Type: Literal["SNS", "SQS"]
+
+
+class FunctionEventInvokeConfiguration(BaseModel):
+    DestinationConfig: Optional[PassThrough]
+    MaximumEventAgeInSeconds: Optional[int]
+    MaximumRetryAttempts: Optional[int]
+
+
+class FunctionEvent(BaseModel):
+    Properties: Any
```

Comment:
Missing types. Also same as comment above; should couple both `Properties` and `Type` so that each model only has a single `Type`.

Reply:
Will do this in a separate PR. It's a huge change.